### DISCUSSION TITLE
Why does Parcel 2 support only Vue 3 SFCs and not Vue 2?

### DISCUSSION THREAD

**Discussion by ghost** [parcel-bundler/parcel#6705] [Q&A]
Upgrading Vue 2 to Vue 3 right now seems to be impracticable. So why does Parcel 2 **require** Vue 3?

The following tutorial shows Vue 2 with an SFC compiling and running just fine in Parcel 1 (I just tried it -- Parcel even auto-installs @vue/component-compiler-utils and vue-template-compiler):
https://scotch.io/tutorials/using-parcel-in-a-vuejs-app

But the Parcel 2 docs state:
> Note that Parcel does not support using SFCs with Vue 2, you must use Vue 3 beta or later.
(https://v2.parceljs.org/languages/vue/)

Why can Parcel 2 not use the vue-template-compiler to compile Vue 2 SFCs -- or be configured to do so?

**Comment by devongovett** [comment]
I think there was some confusion about whether Vue 3 was released or not. From the responses to my tweet, it sounds like it's kind of but not really released. https://twitter.com/devongovett/status/1425182820197036034

It might be possible to auto detect the Vue version and switch to the old compiler, or maybe we will need a separate Parcel plugin. Regardless, there is no technical reason it just hasn't been done yet. If you want to help out with that we'd be happy to help guide you.

**Reply by ghost** [reply]
Man, I just want to find something to replace webpack in Rails 6. Webpack(er) sucks. It is slow, it is painful and brittle, horribly documented. I've got a Rails 6.1 app with Vue 2, using Buefy, which uses Vue 2 with render(createElement) functions, and a whole bunch of stuff that breaks with Vue 3 Proxy objects. Yeah, I tried using @vue/compat -- that's broken, too!!! Parcel would be awesome, if only it would work. I'm trying to fork the great Breakfast gem, and now have it working in Rails (hooray!) but Parcel V1 is returning this weird error:

```js
Uncaught SyntaxError: Illegal return statement
```
Parcel's generated code looks like the following:
```
...
  if (cached && cached.hot && cached.hot._acceptCallbacks.length) {
    cached.hot._acceptCallbacks.forEach(function (cb) {
      cb();
    });

    return true;
  }
}
},{}]},{},["../../node_modules/parcel/src/builtins/hmr-runtime.js"], null)
//# sourceMappingURL=/entrypoints/landing.js.mapperCase();
        if (options.type === 'GET' && options.data) {
          if (options.url.indexOf('?') < 0) {
            options.url += '?' + options.data;
          } else {
            options.url += '&' + options.data;
          }
        }
        if (AcceptHeaders[options.dataType] == null) {
          options.dataType = '*';
        }
        options.accept = AcceptHeaders[options.dataType];
        if (options.dataType !== '*') {
          options.accept += ', */*; q=0.01';
        }
        return options; // <-- THIS IS OUTSIDE A FUNCTION!!!
      };
...
```
So yeah, it looks like Parcel V1 is totally broken/deprecated, and V2 totally does not work with Vue 2, which I must have (really, really wish I could update to Vue 3 but about to put that on the back burner for about 6 months).

Also, Parcel V1 does not seem to generate a manifest, and minify is totally broken in it, so, yeah, if there's a way to get Parcel V2 working with a Vue 2 app, I'd love it.

What can I do to help?

**Reply by ghost** [reply]
Here's the gem I'm trying to build off of Breakfast: https://github.com/dra11y/parcel-rails
